### ISSUE TITLE
[CI] EnrichIT testListeningRatio failing

### ISSUE THREAD

**Issue by benwtrent** [elastic/elasticsearch#109816]
**Build scan:**
https://gradle-enterprise.elastic.co/s/gg6slpy3q6pfo/tests/:x-pack:plugin:esql:internalClusterTest/org.elasticsearch.xpack.esql.action.EnrichIT/testListeningRatio

**Reproduction line:**
```
./gradlew ':x-pack:plugin:esql:internalClusterTest' --tests "org.elasticsearch.xpack.esql.action.EnrichIT.testListeningRatio" -Dtests.seed=DF4CDC0FFA0D2B5A -Dtests.locale=nn-NO -Dtests.timezone=BET -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.xpack.esql.action.EnrichIT#testListeningRatio`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testListeningRatio'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.action.EnrichIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure excerpt:**
```
org.elasticsearch.ResourceNotFoundException: sink exchanger for id [X-0HDkoMR1WQ6Vf59zXpFQ:23] doesn't exist

  at __randomizedtesting.SeedInfo.seed([DF4CDC0FFA0D2B5A:2CED2CE9A0414374]:0)
  at org.elasticsearch.compute.operator.exchange.ExchangeService.getSinkHandler(ExchangeService.java:131)
  at org.elasticsearch.xpack.esql.plugin.ComputeService.runComputeOnDataNode(ComputeService.java:714)
  at org.elasticsearch.xpack.esql.plugin.ComputeService$DataNodeRequestHandler.messageReceived(ComputeService.java:787)
  at org.elasticsearch.xpack.esql.plugin.ComputeService$DataNodeRequestHandler.messageReceived(ComputeService.java:754)
  at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:75)
  at org.elasticsearch.transport.InboundHandler.doHandleRequest(InboundHandler.java:287)
  at org.elasticsearch.transport.InboundHandler$1.doRun(InboundHandler.java:300)
  at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
  at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
  at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
  at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
  at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
  at java.lang.Thread.run(Thread.java:1570)

```

**Comment by benwtrent** [comment]
It doesn't reproduce reliably, but looking at the test history, it definitely seems flaky. If this continues to fail, we should mute.

**Comment by nik9000** [comment]
@dnhatn, another one of these `sink exchanger for id [XXXX] doesn't exist` tests.

**Comment by dnhatn** [comment]
@nik9000 Yes, there are some issues with the exchange timeout. I will spend some time on it.

**Comment by alex-spies** [comment]
Got another one like that for `testTopN`: https://gradle-enterprise.elastic.co/s/cgidqjkhcriq2/tests/task/:x-pack:plugin:esql:internalClusterTest/details/org.elasticsearch.xpack.esql.action.EnrichIT/testTopN

**Comment by dnhatn** [comment]
Fixed in #110450
